cola/widgets/startup.py

```diff
@@ -265,16 +336,397 @@ def handle_broken_repo(self, index):
+
+    def clear_default_item(self, _item):
+        context = self.context
+        cmds.do(cmds.SetDefaultRepo, context, None)
+        self.refresh()
+
+    def rename_repo(self):
+        index = self.currentIndex()
+        self.edit(index)
+
+    def accept_repo(self):
+        self.apply_fn(self.accept_item)
+
+    def accept_item(self, _item):
+        if self.state() & self.EditingState:
+            current_index = self.currentIndex()
+            widget = self.indexWidget(current_index)
+            if widget:
+                self.commitData(widget)
+            self.closePersistentEditor(current_index)
+            self.refresh()
+        else:
+            self.open_selected_repo()
+
+    def open_new_repo(self):
+        context = self.context
+        self.apply_fn(lambda item: cmds.do(cmds.OpenNewRepo, context, item.path))
+
+    def launch_editor(self):
+        context = self.context
+        self.apply_fn(lambda item: cmds.do(cmds.Edit, context, [item.path]))
+
+    def launch_terminal(self):
+        context = self.context
+        self.apply_fn(lambda item: cmds.do(cmds.LaunchTerminal, context, item.path))
+
+    def delete_recent(self):
+        """Removes a selected repo item from the recent list"""
+        cmd = cmds.RemoveRecent
+        self.delete(cmd)
+
+    def delete_bookmark(self):
+        """Removes a selected repo item from the bookmark list"""
+        cmd = cmds.RemoveBookmark
+        self.delete(cmd)
+
+    def delete(self, cmd):
+        item = self.selected_item()
+        context = self.context
+        if not item:
+            return
+        ok, _, _, _ = cmds.do(cmd, context, item.path, item.name, icon=icons.discard())
+        if ok:
+            self.refresh()
+
+    def remove_missing(self):
+        """Remove missing entries from the favorites/recent file list"""
+        settings = self.context.settings
+        settings.remove_missing_recent()
+        self.refresh()
+
+
+def _disable_rename(_x, _y, _z):
+    return False
+
+
+class BuildItem(object):
+    def __init__(self, context):
+        self.star_icon = icons.star()
+        self.folder_icon = icons.folder()
+        cfg = context.cfg
+        self.default_repo = cfg.get('cola.defaultrepo')
+
+    def get(self, path, name):
+        is_default = self.default_repo == path
+        if is_default:
+            icon = self.star_icon
+        else:
+            icon = self.folder_icon
+        return PromptWidgetItem(path, name, icon, is_default)
+
+
+class PromptWidgetItem(QtGui.QStandardItem):
+    def __init__(self, path, name, icon, is_default):
+        QtGui.QStandardItem.__init__(self, icon, name)
+        self.path = path
+        self.is_default = is_default
+        self.name = name
+
+        user_role = Qt.UserRole
+        self.setEditable(False)
+        self.setData(path, user_role)
+        self.setIcon(icon)
+        self.setText(name)
+        self.setToolTip(path)
+        self.setFlags(self.flags() | Qt.ItemIsEditable)
```

Comment:
Is it better to use `setEditable(bool)` rather than using `setFlags()`?

Reply:
Also.. it looks like we call `setEditable(False)` right above here...

Reply:
Oh, using `setEditable(bool)` is enough and `setFlags` is overload.
I'm really sorry.